### PR TITLE
Add crispector

### PR DESCRIPTION
**Pull request by iamit87**
new SW pipeline for an accurate estimation of off-target editing activity from comparative NGS data.
First commit to a new conda-package.
----

> Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences**
      (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

> Everyone has access to the following BiocondaBot commands, which can be given in a comment:
>
>  * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
>  * `@BiocondaBot please add label` will add the `please review & merge` label.
>  * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.
>
> For members of the Bioconda project, the following command is also available:
>
>  * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.
>
> If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

### CODE REVIEW

File: recipes/crispector/build.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+$PYTHON setup.py install --single-version-externally-managed --record=rec.txt
```
Comment (marcelm) [inline recipes/crispector/build.sh]: Please remove this build script and add this to the build section instead (avoids an additional file, also pip install is the recommended way to install Python packages):
```
script: python -m pip install . --no-deps -vv
```

File: recipes/crispector/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  version: "1.0.0"
+
+build:
+  number: 0
+  entry_points:
+    - crispector = crispector.arg_parser:cli
+  noarch: generic
+
+source:
+  url: https://github.com/YakhiniGroup/crispector/archive/v1.0.0b.tar.gz
```
Comment (marcelm) [inline recipes/crispector/meta.yaml]: Is it correct that the version for the archive (1.0.0b) is different from the package version (without the "b")?

File: recipes/crispector/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  sha256: 3dcdad7b4a4dd317a1137186be98376b367ebb20ce7e56e0bb58e1469a04bcde
+
+requirements:
+  host:
+    - python >=3.6
+    - setuptools
+    - biopython 1.74
```
Comment (marcelm) [inline recipes/crispector/meta.yaml]: This dependency and the following ones should be removed from the `host` dependencies. They are not required at install time when using `pip install --no-deps`. But you should add `pip` as a host dependency.

File: recipes/crispector/build.sh (hunk 1 above)
Comment (iamit87) [reply recipes/crispector/build.sh]: Thanks.

File: recipes/crispector/meta.yaml (hunk 2 above)
Comment (iamit87) [reply recipes/crispector/meta.yaml]: No. corrected.

File: recipes/crispector/meta.yaml (hunk 3 above)
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Should I remove all "host:" dependencies and keep just "run:" section?

Comment (marcelm) [reply recipes/crispector/meta.yaml]: No, you need to keep `python`, `setuptools` and `pip`.

File: recipes/crispector/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - jinja2
+
+test:
+  commands:
+    - crispector -h
+
+about:
+  home: https://github.com/YakhiniGroup/crispector
+  license: Partners
+  license_file: LICENSE
```
Comment (marcelm) [inline recipes/crispector/meta.yaml]: The license [explicitly forbids re-distribution](https://github.com/YakhiniGroup/crispector/blob/275fd375cb05b44670ee27d4f44a6b0426ff32a3/LICENSE#L23):
> ... You may not copy or distribute CRISPECTOR and its code in any form.

Are you the copyright holder for the tool? If so, you are of course not bound by those license terms and can just grant Bioconda permission to distribute the tool. To do this properly, I suggest that you
- add a comment to the PR stating that you grant permission to distribute Crispector,
- add a comment to the recipe that links to the PR comment, something like: "permission to distribute the software has been granted by the author at (URL)",
- and optionally, to make it a bit clearer that you’re responsible for the recipe, add yourself as maintainer by adding this section to the recipe (at the bottom):
```
extra:
  recipe-maintainers:
    - iamit87
```
Alternatively, you could of course switch to an Open Source license :-).

If you’re not the copyright holder (or whatever the equivalent in your legislation), you need to get permission first.

File: recipes/crispector/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  script: python -m pip install . --no-deps -vv
+
+
+source:
+  url: https://github.com/YakhiniGroup/crispector/archive/v1.0.2b4.tar.gz
+  sha256: c074d02083818a088b12ac958db97b89d6d74fea6209b674e2f5a016b326113f
+
+requirements:
+  host:
+    - python 3.7
```
[the remark this answers is not in the document]
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Yes, I'll change it to 3.7. 3.6 had a bug with click, so I had to upgrade to 3.7

**Comment by iamit87** [comment]
We grant permission to Bioconda to distribute CRISPECTOR under the end user license agreement

File: recipes/crispector/meta.yaml (hunk 4 above)
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Hi,
Thanks for the detailed response! 
I'm the author of CRISPECTOR.
 - I added myself as the recipe-maintainers.
 - I added a comment in this PR which grant distribute permissions.
-  I added reference to this PR in the README file.

File: recipes/crispector/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+test:
+  commands:
+    - crispector -h
+
+about:
+  home: https://github.com/YakhiniGroup/crispector
+  license: Partners
+  license_file: LICENSE
+  summary: "Accurate estimation of off-target editing activity from comparative NGS data"
+
```
[the remark this answers is not in the document]
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Removed

File: recipes/crispector/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package:
```
[the remark this answers is not in the document]
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Thanks! Good practice...

File: recipes/crispector/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  number: 0
+  entry_points:
+    - crispector = crispector.cli:main
+  noarch: generic
+  script: python -m pip install . --no-deps -vv
+
+
+source:
+  url: https://github.com/YakhiniGroup/crispector/archive/v1.0.2b4.tar.gz
```
[the remark this answers is not in the document]
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Done.

File: recipes/crispector/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+package:
+  name: crispector
+  version: "{{ version }}"
+
+build:
+  number: 0
+  entry_points:
+    - crispector = crispector.cli:main
+  noarch: generic
```
Comment (marcelm) [inline recipes/crispector/meta.yaml]: Sorry I haven’t seen this in the previous round: This should be `noarch: python`.

File: recipes/crispector/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    - matplotlib-base >=3.1.2
+    - scipy >=1.2.1
+    - numpy >=1.12.1
+    - seaborn >=0.9.0
+    - plotly >=4.3.0
+    - pandas >=0.24.2
+    - statsmodels
+    - jinja2
+
+extra:
```
Comment (marcelm) [inline recipes/crispector/meta.yaml]: The `extra:` section should be the last in the recipe (I don’t think this is documented anywhere, but it’s nicer to be consistent with the other recipes).

File: recipes/crispector/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    - statsmodels
+    - jinja2
+
+test:
+  commands:
+    - crispector -h
+
+about:
+  home: https://github.com/YakhiniGroup/crispector
+  license: Partners
```
[the remark this answers is not in the document]
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Done.

File: recipes/crispector/meta.yaml (hunk 10 above)
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Done

File: recipes/crispector/meta.yaml (hunk 9 above)
Comment (iamit87) [reply recipes/crispector/meta.yaml]: Done

**Comment by iamit87** [comment]
@BiocondaBot please add label

**Comment by marcelm** [comment]
Looks good now! I added a comment about the license exception. Feel free to merge when the tests have passed.

**Comment by iamit87** [comment]
@BiocondaBot please merge

**Pull request merged**